src/sentry/api/endpoints/seer_rpc.py

```diff
@@ -0,0 +1,170 @@
+    signature_input = b"%s:%s" % (
+        url.encode("utf8"),
+        body,
+    )
+
+    for key in settings.SEER_RPC_SHARED_SECRET:
+        computed = hmac.new(key.encode("utf-8"), signature_input, hashlib.sha256).hexdigest()
+        is_valid = hmac.compare_digest(computed.encode("utf-8"), signature_data.encode("utf-8"))
+        if is_valid:
+            return True
+
+    return False
+
+
+@AuthenticationSiloLimit(SiloMode.CONTROL, SiloMode.REGION)
+class SeerRpcSignatureAuthentication(StandardAuthentication):
+    """
+    Authentication for seer RPC requests.
+    Requests are sent with an HMAC signed by a shared private key.
+    """
+
+    token_name = b"rpcsignature"
+
+    def accepts_auth(self, auth: List[bytes]) -> bool:
+        if not auth or len(auth) < 2:
+            return False
+        return auth[0].lower() == self.token_name
+
+    def authenticate_token(self, request: Request, token: str) -> Tuple[Any, Any]:
+        if not compare_signature(request.path_info, request.body, token):
+            raise AuthenticationFailed("Invalid signature")
+
+        with configure_scope() as scope:
+            scope.set_tag("rpc_auth", True)
+
+        return (AnonymousUser(), token)
+
+
+@all_silo_endpoint
+class SeerRpcServiceEndpoint(Endpoint):
+    """
+    RPC endpoint for seer microservice to call. Authenticated with a shared secret.
+    """
+
+    publish_status = {
+        "POST": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.ML_AI
+    authentication_classes = (SeerRpcSignatureAuthentication,)
+    permission_classes = ()
+    enforce_rate_limit = False
+
+    def _is_authorized(self, request: Request) -> bool:
+        if request.auth and isinstance(
+            request.successful_authenticator, SeerRpcSignatureAuthentication
+        ):
+            return True
+        return False
+
+    def _dispatch_to_local_method(self, method_name: str, arguments: Dict[str, Any]) -> Any:
+        if method_name not in seer_method_registry:
+            raise RpcResolutionException(f"Unknown method {method_name}")
+        method = seer_method_registry[method_name]
+        return method(**arguments)
+
+    def post(self, request: Request, method_name: str) -> Response:
+        if not self._is_authorized(request):
+            raise PermissionDenied
+
+        try:
+            arguments: Dict[str, Any] = request.data["args"]
+        except KeyError as e:
+            raise ParseError from e
+        if not isinstance(arguments, dict):
+            raise ParseError
+
+        try:
+            result = self._dispatch_to_local_method(method_name, arguments)
+        except RpcResolutionException as e:
+            capture_exception()
+            raise NotFound from e
+        except SerializableFunctionValueException as e:
+            capture_exception()
+            raise ParseError from e
+        except Exception as e:
+            # Produce more detailed log
+            if in_test_environment():
```

Comment:
Why only in pytest, rather than all non-prod?

```suggestion
            if settings.DEBUG:
```